source/time_tables/departure_boards.cpp

```diff
@@ -44,6 +44,28 @@ namespace pt = boost::posix_time;
 
 namespace navitia { namespace timetables {
 
+static bool is_last_stoptime(const nt::StopTime* stop_time, const nt::StopPoint* stp){
+    return stop_time->vehicle_journey
+            && stop_time->vehicle_journey->route
+            && stop_time->vehicle_journey->route->destination
+            && ! stop_time->vehicle_journey->stop_time_list.empty()
+            && stop_time->vehicle_journey->stop_time_list.back().stop_point == stp;
+}
+
+static bool is_partial_terminus(const PbCreator& pb_creator,
```

Comment:
`is_flagged_partial_terminus` would be a better name, as it sometimes is a partial terminus, but you don't flag it that way as you don't have times. (or maybe outsource the `! stop_times.empty()` test)